fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -188,12 +193,58 @@ private void incrementCallCount(FDBStoreTimer.Counts counter) {
         }
     }
 
+    public static boolean isSegmentInfo(String name) {
+        return name.endsWith(SI_EXTENSION);
+    }
+
+    public static boolean isCompoundFile(String name) {
+        return name.endsWith(DATA_EXTENSION);
+    }
+
+    public static boolean isEntriesFile(String name) {
+        return name.endsWith(ENTRIES_EXTENSION);
+    }
+
+    public static String convertToDataFile(String name) {
+        if (isSegmentInfo(name)) {
+            return name.substring(0, name.length() - 2) + DATA_EXTENSION;
+        } else if (isEntriesFile(name)) {
+            return name.substring(0, name.length() - 3) + DATA_EXTENSION;
+        } else {
+            return name;
+        }
+
+    }
+
     /**
      * Puts a file reference in the meta subspace and in the cache under the given name.
      * @param name name for the file reference
      * @param reference the file reference being inserted
      */
-    public void writeFDBLuceneFileReference(@Nonnull final String name, @Nonnull final FDBLuceneFileReference reference) {
+    public void writeFDBLuceneFileReference(@Nonnull String name, @Nonnull FDBLuceneFileReference reference) {
```

Comment:
I think I asked this question in a different PR previously. One side effect of this optimization is we have to read the reference before writing it to persist the segmentInfo and entries. I also think it is worthy to do so considering what we can save from reduced R/W to ```si``` and ```cfe``` intuitively. I am just wondering whether this cost is considered and did we evaluate this tradeoff quantitatively?

Reply:
There's a new test in the LuceneIndex that checks how many times we are checking. I think we can check in the directory that the files exist, lets put a test in the DirectoryTest for that. And validate the contents of the Lucenereference

Reply:
@sophiabuell is this similar thing as what I am suggesting in the comment below, that to have a test to verify no ```si``` or ```cfe``` files are written to DATA_SUBSPACE, and only ```cfs``` is existing?

Reply:
One thing to consider is that this is a RYW Key Value Store...

Reply:
@jleach4 yeah that is right. But I guess we know the keys in Foundation for the target files right?